backend-services/voting-app/src/main/java/org/cardano/foundation/voting/service/auth/web3/KeriWeb3Filter.java

```diff
@@ -125,53 +126,40 @@ protected void doFilterInternal(HttpServletRequest req,
 
         // Step 1: Check if OOBI is already registered
         Either<Problem, String> oobiCheckResult = keriVerificationClient.getOOBI(headerOobi, 1);
-        if (oobiCheckResult.isLeft()) {
-            sendBackProblem(objectMapper, res, oobiCheckResult.getLeft());
-            return;
-        }
 
-        // Log if OOBI is registered or not
-        log.info("OOBI status: {}", oobiCheckResult.get());
+        if (oobiCheckResult.isLeft()) {
+            log.info("OOBI not registered yet: {}", headerOobi);
+            // Step 2: Register OOBI if not already registered
+            val oobiRegistrationResultE = keriVerificationClient.registerOOBI(headerOobi);
 
-        if (oobiCheckResult.isRight()) {
-            log.info("OOBI already registered: {}", oobiCheckResult);
-            Either<Problem, Boolean> verificationResult = keriVerificationClient.verifySignature(headerAid, headerSignature, headerSignedJson);
+            if (oobiRegistrationResultE.isLeft()) {
+                sendBackProblem(objectMapper, res, oobiRegistrationResultE.getLeft());
+                return;
+            }
 
-            if (verificationResult.isEmpty()) {
-                val problem = Problem.builder()
-                        .withTitle("KERI_SIGNATURE_VERIFICATION_FAILED")
-                        .withDetail("Unable to verify KERI header signature, reason: " + verificationResult.swap().get().getDetail())
-                        .withStatus(BAD_REQUEST)
-                        .build();
+            log.info("OOBI registered successfully: {}", headerOobi);
 
-                sendBackProblem(objectMapper, res, problem);
+            // Step 3: Attempt to verify OOBI registration up to 60 times
+            val oobiFetchResultE = keriVerificationClient.getOOBI(headerOobi, 60);
+            if (oobiFetchResultE.isLeft()) {
+                sendBackProblem(objectMapper, res, oobiFetchResultE.getLeft());
                 return;
             }
         }
 
-        log.info("OOBI not registered yet: {}", headerOobi);
-        // Step 2: Register OOBI if not already registered
-        val oobiRegistrationResultE = keriVerificationClient.registerOOBI(headerOobi);
-
-        if (oobiRegistrationResultE.isLeft()) {
-            sendBackProblem(objectMapper, res, oobiRegistrationResultE.getLeft());
+        // Step 1.1:Update key state
+        Either<Problem, Boolean> keyStateUpdateResult = keriVerificationClient.updateAndVerifyKeyState(headerAid, 60);
```

Comment:
perhaps be good to add a note or todo here that this is a stop gap solution while KERI watchers are being developed